src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/18.diff.sql

```diff
@@ -0,0 +1,297 @@
+﻿/*************************************************************
+    Create partition function, scheme and clustered index to migrate data.
+**************************************************************/
+
+IF NOT EXISTS(SELECT * FROM sys.partition_functions WHERE  name = 'PartitionFunction_ResourceChangeData_Timestamp')
+BEGIN            
+    -- Partition function for the ResourceChangeData table.
+    -- It is not a fixed-sized partition. It is a sliding window partition.
+    -- Adding a range right partition function on a timestamp column. 
+    -- Range right means that the actual boundary value belongs to its right partition, 
+    -- it is the first value in the right partition.
+    CREATE PARTITION FUNCTION PartitionFunction_ResourceChangeData_Timestamp (datetime2(7))
+        AS RANGE RIGHT FOR VALUES('1970-01-01T00:00:00.0000000');
+END;
+
+IF NOT EXISTS(SELECT * FROM sys.partition_schemes WHERE name = 'PartitionScheme_ResourceChangeData_Timestamp')
+BEGIN    
+    -- Partition scheme which uses a partition function called PartitionFunction_ResourceChangeData_Timestamp, 
+    -- and places partitions on the PRIMARY filegroup.
+    CREATE PARTITION SCHEME PartitionScheme_ResourceChangeData_Timestamp 
+        AS PARTITION PartitionFunction_ResourceChangeData_Timestamp ALL TO([PRIMARY]);
+END; 
+
+IF (EXISTS(SELECT * FROM sys.partition_functions WHERE  name = 'PartitionFunction_ResourceChangeData_Timestamp')
+	AND EXISTS(SELECT * FROM sys.partition_schemes WHERE name = 'PartitionScheme_ResourceChangeData_Timestamp'))
+BEGIN
+   -- Creates initial partitions based on default 48-hour retention period.
+    DECLARE @numberOfPartitions int = 48;
+    DECLARE @rightPartitionBoundary datetime2(7);
+    DECLARE @currentDateTime datetime2(7) = sysutcdatetime();
+		
+	-- There will be 51 partition boundaries and 52 partitions, 48 partitions for history, 
+    -- one for the current hour, one for the next hour, and 2 partitions for start and end.
+    WHILE @numberOfPartitions >= -1 
+    BEGIN        
+        -- Rounds the start datetime to the hour.
+        SET @rightPartitionBoundary = DATEADD(hour, DATEDIFF(hour, 0, @currentDateTime) - @numberOfPartitions, 0);
+
+		-- Checks if a partition exists.
+		IF NOT EXISTS (SELECT 1 value FROM sys.partition_range_values AS prv
+						JOIN sys.partition_functions AS pf
+							ON pf.function_id = prv.function_id
+					WHERE pf.name = N'PartitionFunction_ResourceChangeData_Timestamp'
+						and CONVERT(datetime2(7), prv.value, 126) = @rightPartitionBoundary) 
+		BEGIN
+			-- Creates new empty partition by creating new boundary value and specifying NEXT USED file group.
+			ALTER PARTITION SCHEME PartitionScheme_ResourceChangeData_Timestamp NEXT USED [Primary];
+			ALTER PARTITION FUNCTION PartitionFunction_ResourceChangeData_Timestamp() SPLIT RANGE(@rightPartitionBoundary); 
+		END;            
```

Comment:
nit: semicolon(;) is not required

Reply:
Should we create a user story to track this?

Reply:
Thank you Robin for creating a task [#86095](https://microsofthealth.visualstudio.com/Health/_workitems/edit/86095).